File: src/lib/pjecalc/core/dominio/calculo/relatorio/demonstrativo-jr-adapter.ts
/**
 * PJe-Calc v2.15.1 — DemonstrativoJRAdapter (abstract) + nested adapters
 * Porte TS-adaptado de:
 *   br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio.DemonstrativoJRAdapter
 *
 * Ref Java: pjecalc-fonte/.../relatorio/DemonstrativoJRAdapter.java
 *
 * Demonstrativo de verbas do cálculo:
 *   - Por verba: nome, periodo, incidencia, formula, totalDoValorCorrigido
 *   - Por ocorrência da verba: periodo, base, div/mult/qty, dobra, devido,
 *     pago, diferenca, indiceAcumulado, valorCorrigido
 *
 * Subclasses concretas: `DemonstrativoJRAdapterPadrao`, `DemonstrativoJRAdapterScript`.
 */
import type Decimal from 'decimal.js';
import type { Periodo } from '../../../base/comum/periodo';
import { JRAdapter, JRAdapterDataSource } from './jr-adapter';

export abstract class DemonstrativoJRAdapter extends JRAdapter {
  abstract getVerbas(): JRAdapterDataSource<DemonstrativoVerbaAdapter>;
}

export abstract class DemonstrativoOcorrenciaAdapter extends JRAdapter {
  abstract getPeriodo(): Periodo;
  abstract getBase(): Decimal;
  abstract getDivisor(): Decimal;
  abstract getMultiplicador(): Decimal;
  abstract getQuantidade(): Decimal;
  abstract getDobra(): string;
  abstract getDevido(): Decimal;
  abstract getPago(): Decimal;
  abstract getDiferenca(): Decimal;
  abstract getIndiceAcumulado(): Decimal;
  abstract getValorCorrigido(): Decimal;
}

export abstract class DemonstrativoVerbaAdapter extends JRAdapter {
  abstract getNome(): string;
  abstract getPeriodo(): Periodo;
  abstract getIncidencia(): string;
  abstract getComentario(): string;
  abstract getFormula(): string;
  abstract getTotalDoValorCorrigido(): Decimal;
  abstract getOcorrencias(): JRAdapterDataSource<DemonstrativoOcorrenciaAdapter>;
}

File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/dominio/calculo/relatorio/DemonstrativoJRAdapter.java
/*
 * Decompiled with CFR 0.152.
 */
package br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio;

import br.jus.trt8.pjecalc.base.comum.Periodo;
import br.jus.trt8.pjecalc.base.comum.relatorio.JRAdapter;
import br.jus.trt8.pjecalc.base.comum.relatorio.JRAdapterDataSource;
import java.math.BigDecimal;

public abstract class DemonstrativoJRAdapter
extends JRAdapter {
    public abstract JRAdapterDataSource<DemonstrativoVerbaAdapter> getVerbas();

    public abstract class DemonstrativoOcorrenciaAdapter
    extends JRAdapter {
        public abstract Periodo getPeriodo();

        public abstract BigDecimal getBase();

        public abstract BigDecimal getDivisor();

        public abstract BigDecimal getMultiplicador();

        public abstract BigDecimal getQuantidade();

        public abstract String getDobra();

        public abstract BigDecimal getDevido();

        public abstract BigDecimal getPago();

        public abstract BigDecimal getDiferenca();

        public abstract BigDecimal getIndiceAcumulado();

        public abstract BigDecimal getValorCorrigido();
    }

    public abstract class DemonstrativoVerbaAdapter
    extends JRAdapter {
        public abstract String getNome();

        public abstract Periodo getPeriodo();

        public abstract String getIncidencia();

        public abstract String getComentario();

        public abstract String getFormula();

        public abstract BigDecimal getTotalDoValorCorrigido();

        public abstract JRAdapterDataSource<DemonstrativoOcorrenciaAdapter> getOcorrencias();
    }
}

